CMake/FindLibXed.cmake

```diff
@@ -1,4 +1,42 @@
 find_path(LibXed_INCLUDE_DIR xed-interface.h)
 find_library(LibXed_LIBRARY NAMES xed)
 
+if (LibXed_INCLUDE_DIR AND LibXed_LIBRARY)
+  include(CheckCSourceRuns)
+  set(CMAKE_REQUIRED_INCLUDES ${LibXed_INCLUDE_DIR})
+  set(CMAKE_REQUIRED_LIBRARIES ${LibXed_LIBRARY})
+  check_c_source_runs("
+    #include <xed-interface.h>
+    #include <stdlib.h>
+    #include <string.h>
+    int main() {
+      const char* version = xed_get_version();
+      if(strlen(version) < 2) {
+        return -1;
+      }
+      // Format is Vxx.yy*
+      // we're only interested in the major version number
+      return !(atoi(version + 1) >= 10);
+    }"
+    LibXed_GOOD_VERSION)
+  set(CMAKE_REQUIRED_INCLUDES)
+  set(CMAKE_REQUIRED_LIBRARIES)
+
+  if (NOT LibXed_GOOD_VERSION)
+    message(STATUS "LibXed version check failed")
+  endif()
```

Comment:
You need to do one out of:
- completely remove support for system installations of libxed
- only use system libxed if explicitly specified
- make it so that if not LibXED_GOOD_VERSION, we use third-party/, even if LibXed is found

In particular, if libxed.so is present, it currently seems more likely to be a library for the X-Apps Editor (https://github.com/linuxmint/xed) than the intel library

Reply:
In `CMake/FindLibXed.cmake` there is a test-run of the system libxed. If it fails, `LibXed_GOOD_VERSION` will be set accordingly.
`FIND_PACKAGE_HANDLE_STANDARD_ARGS` (line 31) will set `LibXed_FOUND` only if: 
* valid paths were found (`LibXed_LIBRARY` and `LibXed_INCLUDE_DIR` are valid) **and**
* `LibXed_GOOD_VERSION` is valid

If there is another library with the same name on the system, the test-run will fail, `LibXed_GOOD_VERSION` will be set to FALSE and `FIND_PACKAGE_HANDLE_STANDARD_ARGS` will **not** set `LibXed_FOUND`.
So, currently, `LibXed_FOUND` is only set if a system libxed exists and it's valid. If `LibXed_FOUND` is not set, third-party is used. This is the same behavior as libLZ4 and follows your third bullet.